CONTRIBUTING.md

```diff
@@ -1,18 +1,77 @@
-    * Bug fixes and various improvements on existing algorithms.
-    * **Contribute to the Open RL Benchmark**
-        * You could add new algorithms or new games to be featured in the Open RL Benchmark (http://benchmark.cleanrl.dev/)
-        * In this case, please contact me (Costa) directly on slack. I will add you to the CleanRL's Team at Weight and Biases (https://wandb.ai/cleanrl). 
-          So if you would *like*, your experiments can be featured on the Open RL Benchmark (http://benchmark.cleanrl.dev/)
-3. 🎇 Submit an PR and get marged!
+Feel free to open an issue or a Pull Request if you have any questions or suggestions. You can also [join our Discord](https://discord.gg/D6RCjA6sVT) and ask questions there. If you plan to work on an issue, let us know in the issue thread to avoid duplicate work.
 
 Good luck and have fun!
+
+## Dev Setup
+
+```bash
+poetry install
+poetry install -E atari
+poetry install -E pybullet
+```
+
+Then you can run the scripts under the poetry environment in two ways: `poetry run` or `poetry shell`. 
+
+* `poetry run`:
+    By prefixing `poetry run`, your command will run in poetry's virtual environment. For example, try running
+    ```bash
+    poetry run python ppo.py
+    ```
+* `poetry shell`:
+    First, activate the poetry's virtual environment by executing `poetry shell`. Then, the name of the poetry's
+    virtual environment (e.g. `(cleanrl-ghSZGHE3-py3.9)`) should appear in the left side of your shell.
+    Afterwards, you can directly run
+    ```bash
+    python ppo.py
+
+
+## Code Formatting
+
+We use [Pre-commit](https://pre-commit.com/) to sort dependencies, remove unused variables and imports, format code using black, and check word spelling. You can run the following command:
+
+```bash
+poetry run pre-commit run --all-files
+```
+
+## Contributing new algorithms
+
+We welcome the contributions of new algorithms.
+
+**Before opening a pull request**, please open an issue first to discuss with us since this is likely a sizable effort. Once we agree on the plan, feel free to make a PR to include the new algorithm.
+
+To help ease the review process, here is a checklist:
+
+1. **Code style**: Make sure you match the code style used in other implemented algorithms in CleanRL. In particular, `poetry run pre-commit run --all-files` will help auto-format the code.
+2. **Empirical analysis and benchmark**: we adopt a similar guide from [sb3-contrib](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/blob/master/CONTRIBUTING.md) with a bit of our spin. The implemented algorithm should come with **tracked** experiments that
+    * match the reported performance in the paper (if applicable)
+    * match the reported performance in a high-quality reference implementation (SB3, Tianshou, and others) (if applicable).
+    * We should also add documentation on how exactly we want the tracked experiments to be done (i.e., what W&B project? should they capture video recording?)
```

Comment:
This sentence reads like an internal memo.
Should it still be in the final version ?

Reply:
Other than that, it reads good to me.

Reply:
Removed.